src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnectionFactory.cs

```diff
@@ -39,7 +39,7 @@ override protected DbConnectionInternal CreateConnection(DbConnectionOptions opt
             SqlInternalConnection result = null;
             SessionData recoverySessionData = null;
 
-            SqlConnection sqlOwningConnection = owningConnection as SqlConnection;
+            SqlConnection sqlOwningConnection = (SqlConnection)owningConnection;
```

Comment:
Why is this being cast instead of using as ? 
What is the advantage of doing the cast? If a type of `owningConnection` is passed here which fails the cast, then a `InvalidCastException` will prevent execution which in this code can be averted by removing the cast. Ideally a `InvalidCastException` should not occur at all.

I would leave this line as is and get rid of the redundant casts below. 

Even if a `DbConnection` which is not `SqlConnection` is provided, the code will fail to get the connection options and a default connection can still be established. 

The redundant casts in the following lines were bad. They were fixed in version of .Net later than 451 as well. 
Thanks for the changes.